modules/assim.batch/R/pda.define.llik.R

```diff
@@ -0,0 +1,88 @@
+##' Define PDA Likelihood Functions
+##'
+##' @title Define PDA Likelihood Functions
+##' @param all params are the identically named variables in pda.mcmc / pda.emulator
+##'
+##' @return List of likelihood functions, one for each dataset to be assimilated against.
+##'
+##' @author Ryan Kelly
+##' @export
+pda.define.llik.fn <- function(settings) {
+  # Currently just returns a single likelihood, assuming the data are flux NEE.
+  llik.fn <- list()
+  for(i in 1:length(settings$assim.batch$inputs)) {
+    # NEE + heteroskedastic Laplace likelihood
+    if(settings$assim.batch$inputs[[i]]$likelihood == "Laplace") {
+        llik.fn[[i]] <- function(model.out, obs.data, llik.par) {
+          resid <- abs(model.out - obs.data)
+          pos <- (model.out >= 0)
+          LL <- c(dexp(resid[pos], 1/(llik.par[1] + llik.par[2]*model.out[pos]), log=TRUE), 
+                  dexp(resid[!pos],1/(llik.par[1] + llik.par[3]*model.out[!pos]),log=TRUE))
+          return(list(LL=sum(LL,na.rm=TRUE), n=sum(!is.na(LL))))
+        }
+    } else {
+      # Default to Normal(0,1)
+        llik.fn[[i]] <- function(model.out, obs.data, llik.par=1) {
+          LL <- dnorm(x= obs.data, mean=model.out, sd=llik.par, log=TRUE)
```

Comment:
normal-ll is there only for testing purposes, we're currently not using it through the workflow. 

The `inputs` are loaded by the pda.load.data.R function. It only loads NEE and sets the parameters for heteroskedastic Laplacian-ll for now (intercept and slopes) . If someone wants to use other data to assimilate against 1) it should be registered to the DB and 2) aligned with the model output during the load.data step according to this design. That would be where they should set the sd and pass along with `inputs` if they're going to use something other than NEE and heteroskedastic Laplacian-ll. 

If you say we should make normal-ll functional as it is (if someone wants to use normal-ll for NEE for some reason), I can make changes in the pda.load.data accordingly